--- boost_1_63_0/boost/test/impl/unit_test_parameters.ipp.orig	2017-08-09 18:45:45.223055629 +0100
+++ boost_1_63_0/boost/test/impl/unit_test_parameters.ipp	2017-08-09 18:45:48.815050508 +0100
@@ -99,7 +99,7 @@
 
 std::string HELP              = "help";
 std::string USAGE             = "usage";
-std::string VERSION           = "version";
+std::string btrt_VERSION           = "version";
 
 //____________________________________________________________________________//
 
@@ -638,10 +638,10 @@
 
     ///////////////////////////////////////////////
 
-    rt::option version( VERSION, (
+    rt::option version( btrt_VERSION, (
         rt::description = "Prints Boost.Test version and exits."
     ));
-    version.add_cla_id( "--", VERSION, " " );
+    version.add_cla_id( "--", btrt_VERSION, " " );
     store.add( version );
 }
 
@@ -676,7 +676,7 @@
         rt::finalize_arguments( s_parameters_store, s_arguments_store );
 
         // Report help if requested
-        if( runtime_config::get<bool>( VERSION ) ) {
+        if( runtime_config::get<bool>( btrt_VERSION ) ) {
             parser->version( std::cerr );
             BOOST_TEST_I_THROW( framework::nothing_to_test( boost::exit_success ) );
         }
