commit cfc7c8306a61ed84255973ad93aa4322f86a811a
Author: Jonathan Wakely <jwakely@redhat.com>
Date:   Thu Jan 26 21:52:37 2017 +0000

    Add header for serialization::make_array

diff --git a/include/boost/mpi/python/serialize.hpp b/include/boost/mpi/python/serialize.hpp
index 5f9136b..8933b34 100644
--- a/include/boost/mpi/python/serialize.hpp
+++ b/include/boost/mpi/python/serialize.hpp
@@ -36,6 +36,7 @@
 
 #include <boost/serialization/split_free.hpp>
 #include <boost/serialization/array.hpp>
+#include <boost/serialization/array_wrapper.hpp>
 
 #include <boost/assert.hpp>
 
